packages/compass-components/src/hooks/use-confirmation.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, useEffect, useState } from 'react';
+
+import {
+  default as ConfirmationModal,
+  Variant as ModalVariant,
+} from '@leafygreen-ui/confirmation-modal';
+
+export { ModalVariant };
+
+type ConfirmationProperties = {
+  title: string;
+  description: string;
+  buttonText?: string;
+  variant?: ModalVariant;
+};
+
+interface ConfirmationModalContextData {
+  showConfirmation: (props: ConfirmationProperties) => Promise<boolean>;
+}
+
+const ConfirmationModalContext =
+  React.createContext<ConfirmationModalContextData>(
+    {} as ConfirmationModalContextData
+  );
+
+class ShowConfirmationEvent extends Event {
+  constructor(
+    public props: ConfirmationProperties,
+    public callback: (value: boolean) => void = () => {}
+  ) {
+    super('show');
+  }
+}
+class GlobalConfirmation extends EventTarget {
+  showConfirmation(props: ConfirmationProperties) {
+    return new Promise<boolean>((resolve) => {
+      const event = new ShowConfirmationEvent(props, resolve);
+      this.dispatchEvent(event);
+    });
+  }
+}
+const globalConfirmation = new GlobalConfirmation();
+export const showConfirmation =
+  globalConfirmation.showConfirmation.bind(globalConfirmation);
+
+export const ConfirmationModalArea: React.FC = ({ children }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [data, setData] = useState({} as ConfirmationProperties);
+  const [callback, setCallback] = useState<(value: boolean) => void>(() => {});
+
+  useEffect(() => {
+    return () => {
+      setIsOpen(false);
+      setData({} as ConfirmationProperties);
+      setCallback(() => {});
+    };
+  }, []);
+
+  // Event listener to use confirmation modal outside of react
+  useEffect(() => {
+    const listener = (event: ShowConfirmationEvent) => {
+      showConfirmation(event.props).then(event.callback);
+    };
+
+    globalConfirmation.addEventListener('show', listener);
+
+    return () => {
+      globalConfirmation.removeEventListener('show', listener);
+    };
+  }, []);
+
+  const showConfirmation = (props: ConfirmationProperties) => {
+    return new Promise<boolean>((resolve, reject) => {
+      if (isOpen) {
+        reject(new Error('Confirmation modal is already open'));
+      } else {
+        setData(props);
+        setIsOpen(true);
+        setCallback(() => resolve);
+      }
+    });
+  };
+
+  const handleConfirm = () => {
+    setIsOpen(false);
+    callback(true);
+  };
+
+  const handleCancel = () => {
+    setIsOpen(false);
+    callback(false);
+  };
+
+  return (
+    <ConfirmationModalContext.Provider value={{ showConfirmation }}>
+      {children}
+      <ConfirmationModal
+        open={isOpen}
+        title={data.title}
+        variant={data.variant ?? ModalVariant.Default}
+        buttonText={data.buttonText ?? 'Confirm'}
+        onConfirm={handleConfirm}
+        onCancel={handleCancel}
+      >
+        {data.description}
+      </ConfirmationModal>
+    </ConfirmationModalContext.Provider>
+  );
```

Comment:
Something I didn't thought about before and is less critical for toasts, but more critical for this component (as reducer logic can rely on this). While we are still working on unifying frontends between DE and Compass we need to find a way to have these providers in multiple places. One way I thought we can use here is something like this, we check if context already exists and based on this either render provider or just render passed children:

```jsx
const hasParentContext = !!useConfirmationModal();

return hasParentContext
  ? <>{children}</>
  : <ConfirmationModalContext.Provider>{children}<ConfirmationModal ...
```

... and then we'll need to wrap plugin components in this provider too. That way you'll be able to use it seamlessly when plugins are used inside and outside the whole app